modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java

```diff
@@ -412,6 +416,60 @@ public void testFailureDetectionOnNodePing2() throws Exception {
         }
     }
 
+    /**
+     * Checks that node leaves the cluster after lose of outgoing connections.
+     *
+     * @throws Exception If any error occurs.
+     */
+    @Test
+    public void testOutgoingConnectionsFailure() throws Exception {
+        try {
+            final int gridCnt = 4;
+
+            failureDetectionTimeout = 2000;
+
+            startGrids(gridCnt - 1);
+
+            nodeSpi.set(new NetFailureTcpDiscoverySpi());
+
+            startGrid(gridCnt - 1);
+
+            UUID failingNodeId = grid(gridCnt - 1).localNode().id();
+
+            CountDownLatch testWaiter = new CountDownLatch(gridCnt - 1);
```

Comment:
Proper name required.
It waits not for a test.

Reply:
Fixed